tests/shared/src/main/scala/munit/AsyncFixtureFrameworkSuite.scala

```diff
@@ -0,0 +1,58 @@
+package munit
+
+import scala.concurrent.Future
+
+class AsyncFixtureFrameworkSuite extends FunSuite {
+  val failingSetup = FunFixture.async[Unit](
+    _ => Future.failed(new Error("failure in setup")),
+    _ => Future.successful(())
+  )
+
+  val failingTeardown = FunFixture.async[Unit](
+    _ => Future.successful(()),
+    _ => Future.failed(new Error("failure in teardown"))
+  )
+
+  val unitFixture = FunFixture.async[Unit](
+    _ => Future.successful(()),
+    _ => Future.successful(())
+  )
+
+  failingSetup.test("fail when setup fails") { _ =>
+    fail("failing setup did not fail the test")
+  }
+
+  failingTeardown.test("fail when teardown fails") { _ => () }
```

Comment:
Would be good to add another test case like this
```suggestion
  failingTeardown.test("fail when teardown fails") { _ => () }
  failingTeardown.test("fail when body and teardown fail") { _ => fail("boom") }
```